src/base/path.h

```diff
@@ -69,8 +69,7 @@ class Path final : public IStringable
     QString toString() const override;
 
     Path &operator/=(const Path &other);
-    Path &operator+=(const QString &str);
-    Path &operator+=(const std::string &str);
+    Path &operator+=(QStringView str);
```

Comment:
So maybe replace all the similar methods that accept `const QString &` (including constructor) to accept `QStringView` parameter?
The same suggestion is for SettingsStorage/SettingValue.
IIRC, `QStringView` is generally intended to be replacement for any `const QString &` parameter.

Reply:
>So maybe replace all the similar methods that accept const QString & (including constructor) to accept QStringView parameter?
The same suggestion is for SettingsStorage/SettingValue.

I don't think it is a good idea.
`SettingValue` needs to store a `QString` and the most efficient way is to construct it via `QString(const QString &)` or `QString(QString &&)`. Using `QStringView::toString()` will incur an unnecessary deep copy.

>IIRC, QStringView is generally intended to be replacement for any const QString & parameter.

Well, I think it depends on what exactly is happening. If the scenario is like `Path::operator+=()` then it is perfect. Otherwise (such as `SettingValue`), not so great.